### PR TITLE
657 properly set link in types with parameter

### LINKED ISSUE #657: Parameterized type in function spec generates an extra anchor tag
I have something similar to the following defined:
```Elixir
@type t :: term
@opaque finger_tree(t) :: ...

@spec append(finger_tree(t), t) :: finger_tree(t)
def append(arg, y) do
 ...
end
```
When docs are generated for this, the entry for `append` looks like

image

It appears that it is trying to add another link to the entry for type `t`

### PR DESCRIPTION
**Pull request by ymtszw**
Should fix #657

### CODE REVIEW

File: lib/ex_doc/formatter/html/autolink.ex [hunk 1]
```diff
@@ -179,6 +181,13 @@ defmodule ExDoc.Formatter.HTML.Autolink do
   end
   defp strip_parens(string, _), do: string
 
+  defp split_string_to_link(string) do
+    case :binary.match(string, "(") do
```
Comment (josevalim) [inline lib/ex_doc/formatter/html/autolink.ex]: Use :binary.split and then you don't need to use match+String.split_at. :)

**Comment by josevalim** [comment]
I have added one tiny comment and this is good to go! Thank you!

File: lib/ex_doc/formatter/html/autolink.ex (hunk 1 above)
Comment (aYuMatsuzawa) [inline lib/ex_doc/formatter/html/autolink.ex]: 0c851c6 here it is.
I thought it should be OK here to exact-match against 2- or 1-element lists, correct?

Comment (ymtszw) [inline lib/ex_doc/formatter/html/autolink.ex]: ^ sorry, alt account

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**